### ISSUE TITLE
Version range issue with updateSbtClassifiers

### ISSUE THREAD

**Issue by alexarchambault** [alexarchambault/coursier#263]
Quick repro:
- create a `project/build.properties` with

```
sbt.version=0.13.11
```
- create a `project/plugins.sbt` with

```
addSbtPlugin("com.github.ddispaltro" % "sbt-reactjs" % "0.6.8")
```
- create an empty `build.sbt`,
- coursier SBT plugin 1.0.0-M12 enabled globally.

Run:

```
$ sbt updateSbtClassifiers
...
[error] (*:coursierSbtClassifiersResolution) coursier.ResolutionException: Encountered 2 error(s) in dependency resolution:
[error]   org.webjars.npm:amdefine:0.0.4:
[error]     not found:
[error]       https://repo.typesafe.com/typesafe/ivy-releases/org.webjars.npm/amdefine/0.0.4/ivys/ivy.xml
[error]       https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/amdefine/0.0.4/ivys/ivy.xml
[error]       /Users/alexandre/.ivy2/local/org.webjars.npm/amdefine/0.0.4/ivys/ivy.xml
[error]       https://repo1.maven.org/maven2/org/webjars/npm/amdefine/0.0.4/amdefine-0.0.4.pom
[error]       https://oss.sonatype.org/content/repositories/releases/org/webjars/npm/amdefine/0.0.4/amdefine-0.0.4.pom
[error]   org.webjars.npm:amdefine:0.0.4:
[error]     not found:
[error]       https://repo.typesafe.com/typesafe/ivy-releases/org.webjars.npm/amdefine/0.0.4/ivys/ivy.xml
[error]       https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/org.webjars.npm/amdefine/0.0.4/ivys/ivy.xml
[error]       /Users/alexandre/.ivy2/local/org.webjars.npm/amdefine/0.0.4/ivys/ivy.xml
[error]       https://repo1.maven.org/maven2/org/webjars/npm/amdefine/0.0.4/amdefine-0.0.4.pom
[error]       https://oss.sonatype.org/content/repositories/releases/org/webjars/npm/amdefine/0.0.4/amdefine-0.0.4.pom
...
```

(output format slightly different than `1.0.0-M12`, changes to be pushed).

**Comment by alexarchambault** [comment]
cc @nafg

**Comment by alexarchambault** [comment]
Curiously, the error only seems to surface via `updateSbtClassifiers`. In particular, putting in `build.sbt`
```
resolvers += Resolver.sbtPluginRepo("releases")
addSbtPlugin("com.github.ddispaltro" % "sbt-reactjs" % "0.6.8")
```
and running `update` or `updateClassifiers` is fine.

Also, from the command line,
```
$ coursier resolve -t \
    -r "ivy:https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]" \
    "com.github.ddispaltro:sbt-reactjs;scalaVersion=2.10;sbtVersion=0.13:0.6.8"
```
and
```
$ coursier fetch \
    -r "ivy:https://repo.scala-sbt.org/scalasbt/sbt-plugin-releases/[organisation]/[module]/(scala_[scalaVersion]/)(sbt_[sbtVersion]/)[revision]/[type]s/[artifact](-[classifier]).[ext]" \
    "com.github.ddispaltro:sbt-reactjs;scalaVersion=2.10;sbtVersion=0.13:0.6.8"
```
are fine, and find `org.webjars.npm:amdefine:0.1.0`.

This dependency is brought via `org.webjars.npm:amdefine:[0.0.4,)`. `org.webjars.npm:amdefine:0.0.4` cannot be found on Central, but the `0.1.0` version can, and the latter should be used for `[0.0.4,)` (it is, except via `updateSbtClassifiers`).

**Comment by alexarchambault** [comment]
As a quick workaround, putting
```
libraryDependencies += "org.webjars.npm" % "amdefine" % "0.1.0"
```
in `project/plugins.sbt`, makes the resolution less confused about the `[0.0.4,)`, and solves the issue.

**Comment by crypticmind** [comment]
Coursier doesn't seem to support the version range expression. I'm seeing it going to our local repo with this URL:

```
http://nexus/nexus/content/repositories/releases/package/artifact-name/%5B0.0,0.1)/artifact-name-%5B0.0,0.1).pom
```

**Comment by damdev** [comment]
Trying this, I find a problem with the range starting from 0.0 version, for example:

./coursier launch 'org.python:jython-standalone:[0.0,2.7.1)' -M org.python.util.jython

doesn't works, but

./coursier launch 'org.python:jython-standalone:[0.0.1,2.7.1)' -M org.python.util.jython

does.

**Comment by alexarchambault** [comment]
@crypticmind @damdev The null version in the interval seems to be the problem here. I opened https://github.com/alexarchambault/coursier/issues/273 for that. This is a different problem than the one of this issue (which is specific to the `updateSbtClassifier` implementation of the SBT plugin at first).

**Comment by alexarchambault** [comment]
@damdev Now realizing this is actually expected, null versions should be used in opened bounds of intervals only, like `(0.0,2.7.1)` or `(0.0,2.7.1]`.

@crypticmind Are you seeing intervals with zero at a close bound in the POM of one of your dependencies? This is not supposed to happen, asking just in case...

**Comment by crypticmind** [comment]
@alexarchambault I have a transitive dependency with version `[0.0,0.1)`. Are you saying the `[0.0` part is not supposed to happen?
From what I see [here](http://maven.apache.org/enforcer/enforcer-rules/versionRanges.html), this doesn't look like an error to me. Am I missing something?

**Comment by damdev** [comment]
@alexarchambault 

I undersand that the null or zero version should be on a open bound, but this error occurs trying to build an existing configuration with coursier. If you want to provide a smoothly transition between plain sbt and sbt+coursier, the support for this version notation it's a good way to accomplish that.

**Comment by alexarchambault** [comment]
Initial issue fixed since `1.0.0-M14` at least.

**Comment by jaiew** [comment]
We are using sbt 0.13.11 with coursier 1.0.0-RC13 and we get the following error:

```
coursier.ResolutionException: Encountered 1 error(s) in dependency resolution:
    org.webjars.npm:jsonify:[0.0.0,0.1):
        not found:
            /Users/jaiew/.ivy2/local/org.webjars.npm/jsonify/[0.0.0,0.1)/ivys/ivy.xml
            https://repo1.maven.org/maven2/org/webjars/npm/jsonify/%5B0.0.0,0.1)/jsonify-%5B0.0.0,0.1).pom
```

It looks like the version range is still an issue?
